### ISSUE TITLE
Huffmunch decompression incorrect.

### ISSUE THREAD

**Issue by cdoty** [jamestn/cpu6502#4]
The huffmunch decompression routine produces incorrect results under cpu6502.
https://github.com/bbbradsmith/huffmunch

**Comment by visrealm** [comment]
Check out my pull requests. I've fixed a number of bugs that could be related.

**Comment by visrealm** [comment]
I tested your project with my latest build of the HBC-56 emulator. I got some output: [https://youtu.be/S2CR4j3f-ew](https://youtu.be/S2CR4j3f-ew)

Is this expected?  It did seem to freeze at the end.

**Comment by visrealm** [comment]
@cdoty Also tested on the real hardware and it seems to be the same now. So, I'd say my pull request fixes should fix the issues you're seeing in my emulator.

[9918 Adventures - HBC-56 - Running on real hardware](https://youtu.be/gS3Ln-ya5FM)

**Comment by cdoty** [comment]
> Is this expected? It did seem to freeze at the end.

Yeah, it's a simple demo project that displays only graphics screens. That is the expected output.

**Comment by visrealm** [comment]
> The problem was the decompression routine generated incorrect data. The latest update fixes the problem.

Great. Thanks for trying it out, Charles.

It was quite strange, I submitted a new pull request yesterday for the broken BIT opcode. Then noticed the 1 Issue in this repo. Had a look and it was related to someone (you) using my HBC-56. I.. was... (pleasantly) surprised. Very cool. :D

I've had a couple of these pull requests for a while, so no sure if this repository is "active". At least the fixes are there if anyone else encounters issues with this repo (I might be the only one).  Or they can use my fork. *shrug*

This issue could be marked closed if the pull requests were accepted.

**Comment by cdoty** [comment]
The debugger in the emulator was nice to use. It made it easy enough to see that the decompression routine wasn't generating the correct data.
